### PR TITLE
Implement MysqlSslWarningSafeHosts to fix #15584

### LINKED ISSUE #15584: Message SSL is not being used even if server is not remote
### Describe the bug
We are running both phpMyAdmin and MariaDB on the same server as the websites. Meaning that MariaDB is not remote, the database connections is done by using "localhost". But still phpMyAdmin display this message in the right column: "Server connection: SSL is not being used"

In phpMyAdmin 4.9.2 the message is in black text color, but in phpMyAdmin 5.1.0-dev it is displayed with red color. It is not fun when customers on shared hosting servers create support tickets and ask about this!

### Expected behavior
When the database server is not remote, the message should not be displayed at all in phpMyAdmin.

### Server configuration
- Operating system: CentOS 7 and 8
- Web server: Apache 2.4.41
- Database version: MariaDB 10.3.20
- PHP version: 7.2.25
- phpMyAdmin version: 4.9.2 and 5.1.0-dev

### Additional context
Please consider to do one of the following so solve this:
1: Remove the message about SSL not being used if the database server is local.
or
2: Give us a configuration setting that we can put in config.inc.php to disable the message, for example something like this:
$cfg['SSLNotBeingUsed_DisableWarning'] = true;

### PR DESCRIPTION
**Pull request by williamdes**
### Description 

Set text to black to avoid users spamming their support.

Fixes #15584

MysqlSslWarningSafeHosts

### Before
image

### After
image

### CODE REVIEW

File: doc/config.rst [hunk 1]
```diff
@@ -1616,6 +1616,26 @@ Generic settings
...
+.. config:option:: $cfg['SslNotUsedSafeHostNames']
+
+    :type: array
+    :default: ``['127.0.0.1', 'localhost']``
+
+    If your setup does not use SSL but is safe because you are using a
+    local connection you can add your hostname or :term:`IP` to the list.
+    You can also remove the default entries to only include yours.
+
+    This check uses the value of :config:option:`$cfg['Servers'][$i]['host']`.
```
Comment (ibennetch) [inline doc/config.rst:1628]: I would add another sentence "This search is case-sensitive and will match the exact string only."

File: doc/config.rst [hunk 2]
```diff
@@ -1616,6 +1616,26 @@ Generic settings
         have to set :config:option:`$cfg['PmaAbsoluteUri']` for correct
         redirection.
 
+.. config:option:: $cfg['SslNotUsedSafeHostNames']
```
Comment (ibennetch) [inline doc/config.rst:1619]: I'm not sure about the directive name, could we make it more clear?

Would MysqlSslWarningSafeHosts be any better?

I don't feel strongly about this, if we can't think of anything better than what you have proposed is fine.

**Review by ibennetch — APPROVED** [review]
Looks great. I have two suggestions but they are minor.

File: doc/config.rst [hunk 3]
```diff
@@ -1616,6 +1616,26 @@ Generic settings
...
+
+    :type: array
+    :default: ``['127.0.0.1', 'localhost']``
+
+    If your setup does not use SSL but is safe because you are using a
+    local connection you can add your hostname or :term:`IP` to the list.
```
Comment (ibennetch) [inline doc/config.rst:1625]: I would add a comma `,` after "local connection"

**Review by ibennetch — APPROVED** [review]

File: doc/config.rst (hunk 3 above)
Comment (ibennetch) [reply doc/config.rst:1625]: How about "local connection or private network, you can add..." ?

**Comment by williamdes** [comment]
I did the changes, feel free to let me know if I can merge this one ;)

File: doc/config.rst (hunk 2 above)
Comment (williamdes) [reply doc/config.rst]: I renamed it, good suggestion :)

File: doc/config.rst (hunk 3 above)
Comment (williamdes) [reply doc/config.rst]: Good catch about private networks, I added it.

**Comment by ibennetch** [comment]
Looks great, you can go ahead and merge it.

**Review by MauricioFauth — APPROVED** [review]

**Pull request merged**